pkg/controllers/sync/dispatch/retain.go

```diff
@@ -48,10 +48,9 @@ func RetainOrMergeClusterFields(
 	// Pass the same ResourceVersion as in the cluster object for update operation, otherwise operation will fail.
 	desiredObj.SetResourceVersion(clusterObj.GetResourceVersion())
 
-	// Retain finalizers and merge annotations since they will typically be set by
+	// Merge annotations since they will typically be set by
 	// controllers in a member cluster.  It is still possible to set the fields
 	// via overrides.
-	desiredObj.SetFinalizers(clusterObj.GetFinalizers())
```

Comment:
Shouldn't this behavior be optional? Retaining finalizers is *usually* the desired behavior; unconditionally removing finalizers without thought could be detrimental to the correctness of controllers in member clusters.